TelemetryTests/TelemetryTests.swift

```diff
@@ -70,30 +74,63 @@ class TelemetryTests: XCTestCase {
         }
     }
 
-    private func storeOnDiskAndUpload(filesOnDisk: Int, expectedUploadCount: Int) {
-        expectedFilesUploaded = expectedUploadCount
+    private func upload(pingType: String) {
+        expectation = expectation(description: "Completed upload")
+        Telemetry.default.scheduleUpload(pingType: pingType)
+        waitForExpectations(timeout: 10.0) { error in
+            if error != nil {
+                print("Test timed out waiting for upload: \(error!)")
+            }
+        }
+    }
 
-        for _ in 0..<filesOnDisk {
+    private func storeOnDiskAndUpload(corePingFilesToWrite: Int) {
+        for _ in 0..<corePingFilesToWrite {
             Telemetry.default.recordSessionStart()
             Telemetry.default.recordSessionEnd()
             Telemetry.default.queue(pingType: CorePingBuilder.PingType)
         }
 
-        let wait = expectation(description: "process async events")
-        XCTWaiter().wait(for: [wait], timeout: 1)
-        print(countFilesOnDisk())
-        XCTAssert(countFilesOnDisk() == filesOnDisk, "Confirm upload file exists")
+        waitForFilesOnDisk(count: corePingFilesToWrite)
+        upload(pingType: CorePingBuilder.PingType)
+    }
 
-        wait.fulfill() // required so it doesn't intefere with waitForExpectations
+    private func waitForFilesOnDisk(count: Int, pingType: String = CorePingBuilder.PingType) {
+        wait()
+        XCTAssert(countFilesOnDisk(forPingType: pingType) == count, "waitForFilesOnDisk matching")
+    }
 
-        expectation = expectation(description: "Completed upload")
-        Telemetry.default.scheduleUpload(pingType: CorePingBuilder.PingType)
+    private func wait() {
+        let wait = expectation(description: "process async events")
+        XCTWaiter().wait(for: [wait], timeout: 0.25)
+        wait.fulfill()
+    }
 
-        waitForExpectations(timeout: 10.0) { error in
-            if error != nil {
-                print("Test timed out waiting for upload: %@", error!)
-            }
+    func testAppEvents() {
+        Telemetry.default.recordEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.foreground, object: TelemetryEventObject.app)
+
+        // Ensure event is stored as expected
```

Comment:
This next block will be removed as part of https://github.com/mozilla-mobile/telemetry-ios/issues/61.